woocommerce/woocommerce
# Analytics API: Customers with non empty passed field as parameter

### Changes proposed in this Pull Request:
The PR adds `filter_empty` parameter to the `wc_analytics/reports/customers` endpoint, that accepts 

```
  'email',
  'name',
  'country',
  'city',
  'state',
  'postcode',
```
enums

Value means that we filter out customers with empty passed value



### Changes proposed in this Pull Request:





Closes #38820



### How to test the changes in this Pull Request:

* Ensure that the Analytics feature is enabled (WooCommerce > Settings > Advanced > Features).
* Create two or more new orders with different customer data. Some with country/state/city some without
* Manually create a customer user account (Users > Add new)
* Use an API client like Postman or Insomnia, or use something like cURL to make requests to the wc-analytics/reports/customers endpoint:
Example request:
```
curl --request GET \
--url 'http://localhost:8086/wp-json/wc-analytics/reports/customers?filter_empty=name,country' \
--header 'Authorization: Basic {your app password}'
```
Ensure that the response has the expected customer records in it, but not any others. Try using other params
